minkindr_gtsam/test/test_minkindr-gtsam.cc

```diff
@@ -683,6 +684,123 @@ TEST(MinkindrGtsamTests, testCubicHermiteQuaternionDerivative) {
+  using namespace gtsam;
+
+  Eigen::Vector3d taVal; taVal.setRandom();
+  Eigen::Vector3d tbVal; tbVal.setRandom();
+  Eigen::Vector3d vaVal; vaVal.setRandom();
+  Eigen::Vector3d vbVal; vbVal.setRandom();
+
+  // Create some values
+  Values values;
+  values.insert(1, taVal);
+  values.insert(2, tbVal);
+  values.insert(3, vaVal);
+  values.insert(4, vbVal);
+
+  EVector3 tA(1), tB(2);
+  EVector3 vA(3), vB(4);
+
+  const double fd_step = 1e-9;
+  const double tolerance = 1e-6;
+
+  {
+    EVector3 interpT1 = kindr::minimal::hermiteInterpolation<Vector3>(tA, vA, tB, vB, 1e-5, 1);
+    SCOPED_TRACE("Testing Expression Jacobians.");
+    testExpressionJacobians(interpT1, values, fd_step, tolerance);
+  }
+  {
+    EVector3 interpT2 = kindr::minimal::hermiteInterpolation<Vector3>(tA, vA, tB, vB, 1.0 - 1e-5, 1);
+    SCOPED_TRACE("Testing Expression Jacobians.");
+    testExpressionJacobians(interpT2, values, fd_step, tolerance);
+  }
+  {
+    EVector3 interpTa = kindr::minimal::hermiteInterpolation<Vector3>(tA, vA, tB, vB, 0.25, 1);
+    SCOPED_TRACE("Testing Expression Jacobians.");
+    testExpressionJacobians(interpTa, values, fd_step, tolerance);
+  }
+  {
+    EVector3 interpTb = kindr::minimal::hermiteInterpolation<Vector3>(tA, vA, tB, vB, 0.75, 1);
+    SCOPED_TRACE("Testing Expression Jacobians.");
+    testExpressionJacobians(interpTb, values, fd_step, tolerance);
+  }
+
+  {
+    EVector3 interpT1 = kindr::minimal::hermiteInterpolationDerivative<Vector3>(tA, vA, tB, vB, 1e-5, 1);
+    SCOPED_TRACE("Testing Expression Jacobians.");
+    testExpressionJacobians(interpT1, values, fd_step, tolerance);
+  }
+  {
+    EVector3 interpT2 = kindr::minimal::hermiteInterpolationDerivative<Vector3>(tA, vA, tB, vB, 1.0 - 1e-5, 1);
+    SCOPED_TRACE("Testing Expression Jacobians.");
+    testExpressionJacobians(interpT2, values, fd_step, tolerance);
+  }
+  {
+    EVector3 interpTa = kindr::minimal::hermiteInterpolationDerivative<Vector3>(tA, vA, tB, vB, 0.25, 1);
+    SCOPED_TRACE("Testing Expression Jacobians.");
+    testExpressionJacobians(interpTa, values, fd_step, tolerance);
+  }
+  {
+    EVector3 interpTb = kindr::minimal::hermiteInterpolationDerivative<Vector3>(tA, vA, tB, vB, 0.75, 1);
+    SCOPED_TRACE("Testing Expression Jacobians.");
+    testExpressionJacobians(interpTb, values, fd_step, tolerance);
+  }
+}
+
+TEST(MinkindrGtsamTests, testCubicHermiteTranslationDerivative) {
+  using namespace gtsam;
+
+
+  Eigen::Vector3d  taVal;  taVal.setRandom();
+  Eigen::Vector3d  tbVal;  tbVal.setRandom();
+  Eigen::Vector3d vaVal; vaVal.setRandom() *= 10.0;
+  Eigen::Vector3d vbVal; vbVal.setRandom() *= 10.0;
+
+  // Create some values
+  Values values;
+  values.insert(1, taVal);
+  values.insert(2, tbVal);
+  values.insert(3, vaVal);
+  values.insert(4, vbVal);
+
+  EVector3 tA(1), tB(2);
+  EVector3 vA(3), vB(4);
+
+  const double fd_step = 1e-9;
+  const double tolerance = 1e-6;
+
+  {
```

Comment:
This could benefit from a comment - it isn't that obvious what you are doing here IMO.